micro-rdk/src/esp32/tcp.rs

```diff
@@ -1,129 +1,519 @@
-use crate::common::conn::server::{AsyncableTcpListener, Http2Connector, OwnedListener};
-use crate::esp32::tls::{Esp32TLS, Esp32TLSStream};
+use crate::common::conn::viam::{HTTP2Stream, IntoHttp2Stream, ViamH2Connector};
 use async_io::Async;
-use futures_lite::{io, ready, AsyncRead, AsyncWrite, Future};
-use hyper::rt;
-use std::fmt::Debug;
-use std::mem::MaybeUninit;
-use std::net::SocketAddr;
+
+use esp_idf_svc::sys::{
+    esp, esp_tls_cfg, esp_tls_cfg_server, esp_tls_conn_destroy, esp_tls_init,
+    esp_tls_role_ESP_TLS_CLIENT, esp_tls_role_ESP_TLS_SERVER, esp_tls_t, EspError,
+};
+use futures_lite::FutureExt;
+use futures_lite::{ready, AsyncRead, AsyncWrite, Future};
+use hyper::{rt, Uri};
+use std::ffi::{c_char, c_void, CString};
+use std::mem::{self, MaybeUninit};
+
+use std::ops::Deref;
+use std::os::fd::AsRawFd;
+use std::pin::Pin;
 use std::sync::Arc;
 use std::{
-    marker::PhantomData,
-    net::{TcpListener, TcpStream},
+    net::TcpStream,
     task::{Context, Poll},
 };
 
-/// Struct to listen for incoming TCP connections
-pub struct Esp32Listener {
-    listener: Arc<TcpListener>,
-    #[allow(dead_code)]
-    addr: SocketAddr,
-    _marker: PhantomData<*const ()>,
-    tls: Option<Box<Esp32TLS>>,
+use super::dtls::{AsyncSSLStream, SSLContext, SSLError};
+
+extern "C" {
+    fn esp_create_mbedtls_handle(
+        hostname: *const c_char,
+        hostlen: i32,
+        cfg: *const c_void,
+        tls: *mut esp_tls_t,
+    ) -> i32;
+}
```

Comment:
is this not importable like the other esp-idf functions?

Reply:
Not exposed, we are hijacking it.